src/core/lib/iomgr/work_serializer.cc

```diff
@@ -33,6 +35,20 @@ struct CallbackWrapper {
   const DebugLocation location;
 };
 
+// First 32 bits indicate ownership of the WorkSerializer, next 32 bits are
```

Comment:
Suggest changing this to use only 8 bits for the ownership and the remaining 56 bits for queue size.  There are likely to be a lot fewer threads competing for ownership than there are entries in the queue.